index.bs

```diff
@@ -6356,51 +6358,48 @@ The following algorithms must only be used on {{ReadableStream}} instances initi
+ 1. Return |byobRequest|.[=ReadableStreamBYOBRequest/[[view]]=].
 </div>
 
 <div algorithm>
- To <dfn export for="ReadableStream">enqueue bytes</dfn> into a {{ReadableStream}} |stream| given a
- [=byte sequence=] |bytes|:
+ To <dfn export for="ReadableStream">enqueue bytes</dfn> into a {{ReadableStream}} |stream| given an
+ {{ArrayBufferView}} |view|:
 
  1. Assert: |stream|.[=ReadableStream/[[controller]]=] [=implements=]
     {{ReadableByteStreamController}}.
- 1. Assert: |bytes|'s [=byte sequence/length=] is positive.
  1. Let |byobRequest| be !
     [$ReadableByteStreamControllerGetBYOBRequest$](|stream|.[=ReadableStream/[[controller]]=]).
- 1. If |byobRequest| is null, or |bytes|'s [=byte sequence/length=] >
-    |byobRequest|.[=ReadableStreamBYOBRequest/[[view]]=]'s [=BufferSource/byte length=], then:
-  1. Let |uint8Array| be the result of [=ArrayBufferView/creating=] a new {{Uint8Array}} from
-     |bytes|.
-  1. Perform ! [$ReadableByteStreamControllerEnqueue$](|stream|.[=ReadableStream/[[controller]]=],
-     |uint8Array|).
- 1. Otherwise,
-  1. Assert: |bytes|'s [=byte sequence/length=] ≤
-     |byobRequest|.[=ReadableStreamBYOBRequest/[[view]]=]'s [=BufferSource/byte length=].
-  1. [=ArrayBufferView/Write=] |bytes| into
-     |byobRequest|.[=ReadableStreamBYOBRequest/[[view]]=].
-
-     Implementations are encouraged to, when possible, do this directly from whatever underlying
-     I/O system is producing |bytes|. In this sense the existence of an intermediate |bytes|
-     variable might end up as spec fiction, if the underlying I/O system can write directly into the
-     backing data block.
-  1. Perform ! [$ReadableByteStreamControllerRespond$](|stream|.[=ReadableStream/[[controller]]=],
-     |bytes|'s [=byte sequence/length=]).
-</div>
-
-When the [=ReadableStream/current requested number of bytes=] is non-null, specifications should
-prefer to use [=ReadableStream/enqueue bytes=] with a [=byte sequence=] whose [=byte
-sequence/length=] equal to (or, if not equal to, less than) that value. Doing so will minimize the
-number of copies involved and will better respect the wishes of stream [=consumers=].
+ 1. If |byobRequest| is non-null, and |view|.\[[ViewedArrayBuffer]].\[[ArrayBufferData]] is the same
+    [=Data Block=] as |byobRequest|.\[[view]].\[[ViewedArrayBuffer]].\[[ArrayBufferData]], then:
+  1. Assert: |view|.\[[ByteOffset]] is |byobRequest|.\[[view]].\[[ByteOffset]].
+  1. Assert: |view|.\[[ByteLength]] ≤ |byobRequest|.\[[view]].\[[ByteLength]].
+     <p class="note">These asserts ensure that the caller does not write outside the requested
+     range in the [=ReadableStream/current BYOB request view=].
+  1. If |view|.\[[ViewedArrayBuffer]] equals |byobRequest|.\[[view]].\[[ViewedArrayBuffer]], then
+     perform ? [$ReadableByteStreamControllerRespond$](|stream|.[=ReadableStream/[[controller]]=],
+     |view|.\[[ByteLength]]).
```

Comment:
Is this needed? Can't we use do `ReadableByteStreamControllerRespondWithNewView` unconditionally? 🤔

Reply:
Yeah, I'm not sure it's actually needed. It felt weird responding with a new view when we didn't have to though... maybe that's just a naming quirk. I.e. we named it `respondWithNewView` instead of `respondWithPossiblyNewView` for brevity.

Reply:
Switched to always use ReadableByteStreamControllerRespondWithNewView